scrapers/distgit.py

```diff
@@ -207,27 +215,45 @@ def _get_repo_info(self, repo_and_commit):
 
         log.debug('Found the cgit URL "{0}" for the commit "{1}" in repo "{2}"'.format(
             url, commit, repo))
-        soup = BeautifulSoup(cgit_result.text, 'html.parser')
         rv['namespace'] = namespace
+
+        # Start parsing the cgit content
+        soup = BeautifulSoup(cgit_result.text, 'html.parser')
+        # Workaround for BS4 in EL7 since `soup.find('th', string=person)` doesn't work in
+        # that environment
+        th_tags = soup.find_all('th')
+        # Temporary variables so that the content can be validated later before being added to rv
+        author = None
+        committer = None
+        for th_tag in th_tags:
+            if th_tag.string == 'author':
+                author = th_tag.next_sibling.string
+            elif th_tag.string == 'committer':
+                committer = th_tag.next_sibling.string
+            elif th_tag.string == 'parent':
+                rv['parent'] = th_tag.next_sibling.find('a').string
+
+            if author and committer and rv['parent']:
+                break
+
         for person in ('author', 'committer'):
```

Comment:
Instead of having a for loop, I could probably just turn this into a function and call it above. What do you think @yashvardhannanavati?

Reply:
Yeap, I agree